### PR TITLE
Add AAD SSO and raw Bearer token authentication methods to Swagger UI

### LINKED ISSUE #433: Implementer SSO i Swagger
Dersom man har lyst til å teste kun backend i dag uten å kjøre opp frontend er man nødt til å logge seg inn på test-alvtime.no og hente ut en bearer token for så å bruke den i Postman e.l. Evt skru av autentisering.

Det er ønskelig å implementere login mot Azure AD via swagger UI, helst med SSO, for å enkelt kunne teste kun backend som innlogget bruker.

### PR DESCRIPTION
**Pull request by FredrikAugust**
Closes #433
Screenshot 2022-04-19 at 15 14 16

Added authentication with AAD SSO and raw bearer tokens (helpful with PATs and AHRE-KETIL LILLEHAGEN).

Also renamed the Swagger schemas to something more useful.

### CODE REVIEW

**Comment by palbock** [comment]
Når jeg prøver å signe inn med AAD så ender jeg bare opp på en blank redirect side
image

File: packages/api/AlvTimeWebApi/Startup.cs [hunk 1]
```diff
@@ -34,14 +36,81 @@ public Startup(IHostEnvironment env)
...
+                contextLifetime: ServiceLifetime.Scoped, optionsLifetime: ServiceLifetime.Scoped);
             services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_3_0);
             services.AddAlvtimeAuthentication(Configuration);
             services.Configure<TimeEntryOptions>(Configuration.GetSection("TimeEntryOptions"));
             services.AddAlvtimeAuthorization();
+
             services.AddSwaggerGen(c =>
             {
-                c.SwaggerDoc("v1", new OpenApiInfo { Title = "Example API", Version = "v1" });
+                c.SwaggerDoc("v1", new OpenApiInfo {Title = "Alvtime API", Version = "v1"});
```
Comment (palbock) [inline packages/api/AlvTimeWebApi/Startup.cs]: Kunne det vært en idè å trekke dette ut i en egen klasse/metode så ikke Startup.cs blir så oppblåst? Litt samme som vi har gjort med services.AddAlvtimeAuthentication og services.AddAlvtimeAuthorization.

Comment (FredrikAugust) [reply packages/api/AlvTimeWebApi/Startup.cs]: Det kunne vært en idé!

Comment (FredrikAugust) [reply packages/api/AlvTimeWebApi/Startup.cs]: Fixed in c960fee

**Comment by FredrikAugust** [comment]
> Når jeg prøver å signe inn med AAD så ender jeg bare opp på en blank redirect side
> image

@palbock90 

I tested this in Safari, FF and Chrome and was not able to reproduce the issue.

Could you check the developer console and see if it prints something?

**Comment by palbock** [comment]
> > Når jeg prøver å signe inn med AAD så ender jeg bare opp på en blank redirect side
> > image
> 
> @palbock90
> 
> I tested this in Safari, FF and Chrome and was not able to reproduce the issue.
> 
> Could you check the developer console and see if it prints something?

@FredrikAugust Seems like its a Firefox issue, works fine in Chrome. Here is error in Firefox:
image

**Comment by FredrikAugust** [comment]
Alright, @palbock90. That's not functionality I've implemented, so I believe the error resides with Swagger.

**Review by palbock — APPROVED** [review]

**Pull request merged**